packaging: make varlink as hard requirement

ffmancera opened the pull request:
Varlink is only used for varlink support so it should not be imported
when doing other operations using nmstatectl.

```
File "/usr/lib/python3.6/site-packages/nmstatectl/nmstatectl.py", line 40, in <module>
  from nmstatectl.nmstate_varlink import start_varlink_server
File "/usr/lib/python3.6/site-packages/nmstatectl/nmstate_varlink.py", line 31, in <module>
  raise libnmError.NmstateDependencyError("python3 varlink module not found")
libnmstate.error.NmstateDependencyError: python3 varlink module not found
```

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

tyll commented on nmstatectl/nmstatectl.py:
```diff
@@ -31,13 +31,18 @@
-from nmstatectl.nmstate_varlink import start_varlink_server
+
+try:
+    from nmstatectl.nmstate_varlink import start_varlink_server
```
if varlink is not available, the help should also not show the varlink support

ffmancera committed:
packaging: make varlink as hard requirement

Nmstatectl is providing an option for running using varlink, therefore
the installation should get everything ready for run instead of
complaining on runtime.

In addition, all the target distributions provides varlink from their
repositories so it will not block the installation.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

tyll commented on packaging/nmstate.spec:
```diff
@@ -35,6 +35,7 @@ Recommends:     NetworkManager-config-server
 Suggests:       NetworkManager-ovs
 Suggests:       NetworkManager-team
 Requires:       python3-nispor
+Requires:       python3dist(varlink)
```
This should not be necessary since requirements.txt are considered by rpmbuild.

ffmancera replied:
Removed, thanks.

tyll commented:
It would be nice to also have CI jobs that check whether the package is installable on all our target distributions. Since the container images include all the anticipated dependencies, it is not tested, there.

cathay4t commented:
> It would be nice to also have CI jobs that check whether the package is installable on all our target distributions. Since the container images include all the anticipated dependencies, it is not tested, there.

That sounds like a job of downstream packager's work.
